app/src/main/java/com/example/caregiver/EditTask.java

```diff
@@ -0,0 +1,167 @@
+package com.example.caregiver;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.util.Log;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EditTask extends AppCompatActivity {
+
+    // fields in layout
+    private EditText taskNameField;
+    private EditText taskNotesField;
+    private EditText caregiveeField;
+    private Spinner roomSpinner;
+
+    private TaskFragment.Task currTask; // current task being edited
+    final DatabaseReference database = FirebaseDatabase.getInstance().getReference();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_task);
+
+        // retrieve info from TaskFragment page intent
+        Intent intent = getIntent();
+        currTask = intent.getParcelableExtra("currTask");
+        String[] caregiveeRooms = intent.getStringArrayExtra("rooms");
+        String caregiveeName = intent.getStringExtra("caregiveeName");
+
+        /* TODO better error handling */
+        // if task is not found, do not display
+        if(currTask == null) return;
+
+        // get all field/spinners
+        taskNameField = findViewById(R.id.taskName);
+        taskNotesField = findViewById(R.id.taskNotes);
+        roomSpinner = findViewById(R.id.taskRoom);
+        caregiveeField = findViewById(R.id.taskCaregivee);
+
+        // populate all fields with task info & create spinner
+        taskNameField.setText(currTask.taskName, TextView.BufferType.EDITABLE);
+        taskNotesField.setText(currTask.taskNote, TextView.BufferType.EDITABLE);
+        caregiveeField.setText(caregiveeName, TextView.BufferType.NORMAL);
+        createSpinner(caregiveeRooms, currTask.room);
+
+        Log.i("INFO", "EditTask:onCreate displayed all task details successfully.");
+
+        /* TODO handle task removal */
+    }
+
+    /**
+     * Initializes room spinner, creates adapter with list of all caregivee rooms
+     * @param allCaregiveeRooms, list of caregivee rooms
+     * @param currRoom, room of current task
+     */
+    protected void createSpinner(String[] allCaregiveeRooms, String currRoom){
+        // set curr room to first entry
+        for(int i = 0; i < allCaregiveeRooms.length; i++){
+            // swap first with curr room
+            if(allCaregiveeRooms[i].equals(currRoom)){
```

Comment:
Hmm, this seems slightly convoluted. Instead of looping through every element in array, can we change this to find the index of the room, and just do a swap? 

```java
int currentRoomIndex = allCaregiveeRooms.indexOf(currRoom);
Collections.swap(allCaregiveeROoms, currentRoomIndex, 0);   
```

Not sure if Collections.swap work for primitives. If it doesn't, feel free to keep your current swap function.

Reply:
I did feel like there should be a simpler way! I removed the loop and used` indexOf` but still had to keep the swapping bit